.github/workflows/ci-cd.yml

```diff
@@ -91,17 +92,10 @@ jobs:
       with:
         python-version: ${{ matrix.python }}
 
-    - name: Install pipenv
-      uses: dschep/install-pipenv-action@v1
-
-    - name: Install Wheel
-      run: pip install wheel
-
-    - name: Build sdist and wheel packages
-      run: python setup.py sdist bdist_wheel
-
-    - name: Push package to Pypi
-      uses: pypa/gh-action-pypi-publish@release/v1
+    - name: Install poetry
+      uses: abatilo/actions-poetry@v2.0.0
       with:
-        user: __token__
-        password: ${{ secrets.pypi_token }}
+        poetry-version: 1.1.13
+
+    - name: Build and push package to PYPI
+      run: poetry run nox -s release -- --username=__token__ --password=${{ secrets.pypi_token }}
```

Comment:
Is this best practices to Adresse the secret this way

Reply:
I removed the password flag, I can see that it could be used locally by accident. Switched to env var only set during exactly the publish invocation. Poetry [config](https://python-poetry.org/docs/repositories/#configuring-credentials) supports keyring, but not sure if that's in the image out of the box and if it fails it will fall back to write it to a file.

Reply:
That is actually better, because we usually only want to release from the CI